bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/LedgerMetadataSerDe.java

```diff
@@ -164,9 +201,112 @@ public LedgerMetadataFormat buildProtoFormat(LedgerMetadata metadata) {
             }
             builder.addSegment(segmentBuilder.build());
         }
-        return builder.build();
+
+        builder.build().writeDelimitedTo(os);
+        return os.toByteArray();
+    }
+
+    private static byte[] serializeVersion2(LedgerMetadata metadata) throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
```

Comment:
don't we need to put os in a try-with-resource block to release os at the end?

Reply:
added, but not strictly needed. there's no resource held here.